App/stores/distanceUnit.tsx

```diff
@@ -0,0 +1,53 @@
+import { AsyncStorage } from 'react-native';
+import React, { useState, useEffect, createContext, FC, useContext } from 'react';
+
+import { i18n } from '../localization';
+
+export type DistanceUnit = 'km' | 'mile';
+type DistanceUnitFormat = 'short' | 'long';
+
+const STORAGE_KEY = 'DISTANCE_UNIT';
+
+type ContextType = {
+  distanceUnit: DistanceUnit;
+  setDistanceUnit: (distanceUnit: DistanceUnit) => void;
+  localizedDistanceUnit: (format: DistanceUnitFormat) => string;
+};
+
+const Context = createContext<ContextType>({
+  distanceUnit: 'km',
+  localizedDistanceUnit: () => '',
+  setDistanceUnit: () => {}
+});
+
+export const DistanceUnitProvider: FC = ({ children }) => {
+  const [distanceUnit, setDistanceUnit] = useState<DistanceUnit>('km');
```

Comment:
The default distanceUnit should be `mile` if i18n's i18n.locale is `en-US`, or `km` in all other cases.